test/e2e/network/netpol/probe.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package netpol
+
+import (
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+)
+
+// ProbeJob packages the data for the input of a pod->pod connectivity probe
+type ProbeJob struct {
+	PodFrom  *Pod
+	PodTo    *Pod
+	FromPort int
+	ToPort   int
+	Protocol v1.Protocol
+}
+
+// ProbeJobResults packages the data for the results of a pod->pod connectivity probe
+type ProbeJobResults struct {
+	Job         *ProbeJob
+	IsConnected bool
+	Err         error
+	Command     string
+}
+
+// ProbePodToPodConnectivity runs a series of probes in kube, and records the results in `testCase.Reachability`
+func ProbePodToPodConnectivity(k8s *Kubernetes, model *Model, testCase *TestCase) {
+	k8s.ClearCache()
+	numberOfWorkers := 30
```

Comment:
ginkgo runs 25 workers in parallel by default
https://github.com/kubernetes/kubernetes/blob/c4ce4206674b1351dcae4d4609e23ca4e0ec0744/hack/ginkgo-e2e.sh#L130-L132

this means that one of those workers will have additional 30 workers, seems is kind of duplication ginkgo.parallel functionality

Reply:
related to above comment - me and Matt will revise the parallelism and update thanks @aojea  good pt :)

Reply:
I thin one worker per NS is going to be fast enough, @eth0xFEED